src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -107,137 +114,263 @@
-                .collect(Collectors.toMap(p -> p.type().getSimpleName(), Function.identity()));
+        private static Map<String, RandomGeneratorProperties> createFactoryMap() {
+            return Map.ofEntries(
+                    entry(SecureRandom.class, "SecureRandom", "Legacy",
+                            0, 0, 0, Integer.MAX_VALUE,
+                            INSTANTIABLE | BYTE_ARRAY_SEED | STOCHASTIC),
+                    entry(Random.class, "Random", "Legacy",
+                            48, 0, 0, 0,
+                            INSTANTIABLE | LONG_SEED),
+                    entry(SplittableRandom.class, "SplittableRandom", "Legacy",
+                            64, 0, 0, 1,
+                            INSTANTIABLE | LONG_SEED),
+                    entry(L32X64MixRandom.class, "L32X64MixRandom", "LXM",
+                            64, 1, 32, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X128MixRandom.class, "L64X128MixRandom", "LXM",
+                            128, 1, 64, 2,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X128StarStarRandom.class, "L64X128StarStarRandom", "LXM",
+                            128, 1, 64, 2,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X256MixRandom.class, "L64X256MixRandom", "LXM",
+                            256, 1, 64, 4,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X1024MixRandom.class, "L64X1024MixRandom", "LXM",
+                            1024, 1, 64, 16,
+                            ALL_CONSTRUCTORS),
+                    entry(L128X128MixRandom.class, "L128X128MixRandom", "LXM",
+                            128, 1, 128, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(L128X256MixRandom.class, "L128X256MixRandom", "LXM",
+                            256, 1, 128, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(L128X1024MixRandom.class, "L128X1024MixRandom", "LXM",
+                            1024, 1, 128, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(Xoroshiro128PlusPlus.class, "Xoroshiro128PlusPlus", "Xoroshiro",
+                            128, 1, 0, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(Xoshiro256PlusPlus.class, "Xoshiro256PlusPlus", "Xoshiro",
+                            256, 1, 0, 3,
+                            ALL_CONSTRUCTORS),
+                    entry(ThreadLocalRandom.class, "ThreadLocalRandom", "Legacy",
+                            64, 0, 0, 1,
+                            0)
+            );
+        }
+
+        private static SimpleImmutableEntry<String, RandomGeneratorProperties>
+        entry(Class<? extends RandomGenerator> rgClass, String name, String group,
+                int i, int j, int k, int equidistribution,
+                int flags) {
+            return new SimpleImmutableEntry<>(name,
```

Comment:
This should use [`Map​.Entry​<…>`] as the return type, which in turn allows using [`Map​::entry​(K, V)`] as the implementation, as neither the `name` key, nor the value is ever `null`:
```suggestion
        private static Map.Entry<String, RandomGeneratorProperties>
        entry(Class<? extends RandomGenerator> rgClass, String name, String group,
                int i, int j, int k, int equidistribution,
                int flags) {
            return Map.entry(name,
```

[`Map​.Entry​<…>`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/util/Map.Entry.html
[`Map​::entry​(K, V)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/util/Map.html#entry(K,V)

Reply:
@ExE-Boss Thanks for the suggestion.